java-frontend/src/main/java/org/sonar/java/SonarComponents.java

```diff
@@ -307,6 +322,31 @@ public File workDir() {
   }
 
   public boolean shouldGenerateUCFG() {
-    return !context.activeRules().findByRepository("SonarSecurityJava").isEmpty();
+    Set<String> activeRuleKeys = context.activeRules().findByRepository(this.ruleRepositoryKey).stream()
+      .map(ActiveRule::ruleKey)
+      .map(RuleKey::rule)
+      .collect(Collectors.toSet());
+    Set<String> securityRuleKeys = getSecurityRuleKeys();
+    activeRuleKeys.retainAll(securityRuleKeys);
+    return !activeRuleKeys.isEmpty();
   }
+
+  public static Set<String> getSecurityRuleKeys() {
+    try {
+      Class<?> javaRulesClass = Class.forName("com.sonar.plugins.security.api.JavaRules");
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Remove this use of dynamic class loading. ([squid:S2658](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2658))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=1966&issues=AWJtT8N4sgPCf6tZNy0m&open=AWJtT8N4sgPCf6tZNy0m)

[ISSUE_KEY]: AWJtT8N4sgPCf6tZNy0m